### DISCUSSION TITLE
Link angle placement

### DISCUSSION THREAD

**Discussion by olicendr** [clientIO/joint#1883] [Q&A]
Hello!
Is it possible to make an angle in the middle of the line, not in the end?
![Untitled](https://user-images.githubusercontent.com/90215998/200809156-b9708a9c-94cd-4b57-841a-bd982ceac835.png)

**Comment by kumilingus** [comment]
No, it's not possible with the current implementation of `manhattan`.

You can try to write a custom router though.
```js
paper.options.defaultRouter = function(vertices, opt, linkView) {
 if (someObstaclesInTheWay(linkView)) {
    // avoid link going through obstacles
    return joint.routers.manhattan.call(this, vertices, opt, linkView);
  }
  /* return the route that breaks exactly in the middle */
  return [/* an array of points */];
}
```

Examples of custom routers:
https://jsfiddle.net/kumilingus/tnuopjah/88/ (this one actually breaks right in the middle)
https://jsfiddle.net/kumilingus/jvdn451y/
